app/src/main/java/com/example/sharewhatyoucanproject/LoginActivity.kt

```diff
@@ -62,170 +56,155 @@ class LoginActivity : AppCompatActivity() {
+                                                        val i = Intent(
+                                                            this,
+                                                            DrawerActivity2::class.java
+                                                        )
+                                                        startActivity(i)
+                                                        finishAffinity()
+                                                    }
+                                                } else {
+                                                    pd.dismiss()
+                                                    Toast.makeText(
+                                                        applicationContext,
+                                                        "Failed " + task3.exception,
+                                                        Toast.LENGTH_SHORT
+                                                    ).show()
+                                                }
+                                            }
+                                        )
+                                } else {
+                                    pd.dismiss()
+                                    Toast.makeText(
+                                        applicationContext,
+                                        "Failed " + task2.exception,
+                                        Toast.LENGTH_SHORT
+                                    )
+                                        .show()
+                                }
+                            }
+                    }
+                } else {
+                    pd.dismiss()
+                    Toast.makeText(applicationContext, "Failed " + it.exception, Toast.LENGTH_SHORT)
+                        .show()
+                }
             }
-        })
+        )
     }
 
-
     fun loginuser() {
-        auth.signInWithEmailAndPassword(
-            email, "Test@123"
-        ).addOnCompleteListener(OnCompleteListener { task ->
-            if (task.isSuccessful) {
-                pd.dismiss()
-                if (type == 1) {
-                    myEdit.putInt("selector", 1)
-                    myEdit.commit()
-                    val i = Intent(this@LoginActivity, DrawerActivity::class.java)
-                    startActivity(i)
-                    finishAffinity()
-                } else if (type == 2) {
-                    myEdit.putInt("selector", 2)
-                    myEdit.commit()
-                    val i = Intent(this@LoginActivity, DrawerActivity2::class.java)
-                    startActivity(i)
-                    finishAffinity()
+        auth.signInWithEmailAndPassword(email, "Test@123").addOnCompleteListener(
+            OnCompleteListener { task ->
+                if (task.isSuccessful) {
+                    pd.dismiss()
+                    if (type == 1) {
```

Comment:
"type"?  Type of what?  Also, what does "1" mean?

Make more self documenting.